src/constants.ts

```diff
@@ -3,6 +3,13 @@ export { CHAIN_IDs, TOKEN_SYMBOLS_MAP } from "@across-protocol/contracts-v2/dist
 
 export const { AddressZero: ZERO_ADDRESS } = ethersConstants;
 
+// 2^96 - 1 is a conservative erc20 max allowance.
+export const MAX_SAFE_ALLOWANCE = "79228162514264337593543950335";
+
+export const MAX_SAFE_JS_INT = "4294967297"; // 2^32 + 1
```

Comment:
IMO I'm not sure we should keep the name `MAX_SAFE_JS_INT` given that `Number.MAX_SAFE_INTEGER` is defined to be `2^53 - 1`

Would it make sense to reference the fact that this would be a max safe integer for an unsigned 32 bit integer?

Reply:
I also wondered if we could use this, and I'd be totally happy to ditch `MAX_SAFE_JS_INT` in favour of `MAX_SAFE_INTEGER`. The only reason I didn't do that is because the UBAFeeCalculator test relies on it, so if we're OK to update that implementation then I think we should do it.

nb. one additional consideration: I noticed that `BigNumber.from(Number.MAX_SAFE_INTEGER)` reverts unless `Number.MAX_SAFE_INTEGER` is cast to a string. We should be fine to handle that, but I thought it was worth mentioning.

Reply:
Strange. Follow up: would it make sense to directly convert this to a BigNumber instead of a string? AFAIK the string representation isn't actually needed and we almost always directly convert this to a BN

Reply:
Per https://github.com/across-protocol/sdk-v2/pull/394, I'll remove the MAX_SAFE_JS_INT entirely :+1: